dpdk, Introduce design doc (#125)
kiagnose/kiagnose: docs/dpdk_checkup_design.md

DPDK Checkup
=

**Authors**: [Edward Haas](https://github.com/eddev), [Orel Misan](https://github.com/orelmisan), [Or Mergi](https://github.com/ormergi), [Ram Lavi](https://github.com/RamLavi).

# Summary

> Provide a functionality to validate DPDK communication going through a virtual machine that runs on a [Kubernetes](https://kubernetes.io/) cluster with [KubeVirt](https://kubevirt.io/).

Reviewer: ```suggestion
Provide a functionality to validate [DPDK](https://www.dpdk.org/) communication going through a virtual machine that runs on a [Kubernetes](https://kubernetes.io/) cluster with [KubeVirt](https://kubevirt.io/).
```

Author: Done

# Motivation

> Configuring DPDK inside a cluster is a multi-step configuration task. A cluster administrator can benefit from a check that can help verify that the overall configuration is optimized for DPDK application.

Reviewer: ```suggestion
Configuring DPDK inside a cluster is a multi-step configuration task. A cluster administrator can benefit from a checkup that can help verify that the overall configuration supports execution of a DPDK application.
```

Author: Like! Done

# Summary

> Provide a functionality to validate dpdk communication going through a virtual machine that runs on a [Kubernetes](https://kubernetes.io/) cluster with [KubeVirt](https://kubevirt.io/)

Reviewer: There is one more instance of "dpdk" in the PR description:
```
This PR introduces a skeleton design doc of the dpdk checkup.
```

## Goals

* Validate DPDK communication going through a VMI, on a predefined network.
* Measure latency of traffic going through a VMI, on a predefined network.

> * Report results (and logs) with the cluster administrator.

Reviewer 2: I am not sure this can be classified as a goal.
We need result reports because we want to _help detect and troubleshoot cluster issues related to high performance VM/s_.

# Proposal

## Definition of Users

> * Kubernetes Cluster Administrators

Reviewer 2: What about K8s namespace administrators?

## User Stories

As a Kubernetes Cluster Administrator I would like to:
1. Check that the cluster network is DPDK-ready by running the DPDK-checkup.

> 2. Optionally provide the maximum accepted latency, such that beyond this value, the check will fail.

Reviewer 2: - Validate that given a namespace in a cluster, with secondary high throughput network, the high performance VM/s can run successfully.
- Validate that there is connectivity between two high performance VM/s on the secondary network.
- Measure the communication latency between the two VM/s, 
- Report the success or failure of the validations, including measured latencies and reasons of failure.
